build/Dockerfile

```diff
@@ -1,26 +1,27 @@
 FROM quay.io/app-sre/boilerplate:image-v2.3.2 AS builder
-
-RUN mkdir -p /workdir
-WORKDIR /workdir
-COPY go.mod go.sum ./
+FROM golang:1.17 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
-COPY . .
-RUN make go-build
-
-####
-FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
-
-ENV USER_UID=1001 \
-    USER_NAME=ocm-agent-operator
-
-COPY --from=builder /workdir/build/_output/bin/* /usr/local/bin/
 
-COPY build/bin /usr/local/bin
-RUN  /usr/local/bin/user_setup
+# Copy the go source
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
 
-ENTRYPOINT ["/usr/local/bin/entrypoint"]
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
 
-USER ${USER_UID}
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
```

Comment:
What is the motivation to switching from `ubi-minimal` to this?

Reply:
Thanks Matt for the review.
This is default Dockerfile and i will addressed this in my latest commit.